synapse/replication/tcp/client.py

```diff
@@ -126,8 +126,9 @@ async def on_rdata(
                 StreamKeyType.ACCOUNT_DATA, token, users=[row.user_id for row in rows]
             )
         elif stream_name == ReceiptsStream.NAME:
+            new_token = self.store.get_max_receipt_stream_id()
             self.notifier.on_new_event(
-                StreamKeyType.RECEIPT, token, rooms=[row.room_id for row in rows]
+                StreamKeyType.RECEIPT, new_token, rooms=[row.room_id for row in rows]
```

Comment:
Is `token` still an `int` in this case?

Reply:
Oh I guess the `token` is the location in the `RDATA` stream?

Reply:
Yup. The "token" here should really be a `PersistedPosition`, since we're saying "this new 'row' was persisted at X"

Reply:
So was this a bug previously? 😄

Reply:
Ah, no. When a stream only has one writer then the "position in the stream" and a "token" are fairly interchangeable (the integer for the persisted position is the same as the integer used for the token that is *after* that position).

Reply:
We're therefore fairly bad at differentiating them in the codebase, as they've historically been the same